pkgs/by-name/po/portaudio-java/package.nix

```
@@ -0,0 +1,99 @@
+{
+  lib,
+  stdenv,
+  portaudio,
+  jdk,
+  cmake,
+  gradle,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "portaudio-java";
+  version = "0-unstable-2023-07-04";
+  src = fetchFromGitHub {
+    owner = "philburk";
+    repo = "portaudio-java";
+    rev = "2ec5cc47d6f8abe85ddb09c34e69342bfe72c60b";
+    hash = "sha256-t+Pqtgstd1uJjvD4GKomZHMeSECNLeQJOrz97o+lV2Q=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gradle
+    jdk
+  ];
+  buildInputs = [ portaudio ];
+
+  # Set up necessary environment variables
+  env = {
+    JAVA_HOME = jdk.home;
+    GRADLE_HOME = gradle;
+  };
+
+  # Fix CMake installation path issue
+  cmakeFlags = [
+    "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}"
+    # Fix gradlew cannot find library issue
+    "-DCMAKE_INSTALL_LIBDIR=lib"
+    "-DCMAKE_BUILD_TYPE=Release"
+  ];
+
+  configurePhase = ''
+    cmake . $cmakeFlags
+  '';
```

```suggestion
  configurePhase = ''
    runHook preConfigure

    cmake . $cmakeFlags

    runHook postConfigure
  '';
```
Similarly, the other overriden phases should run their respective hooks